### DISCUSSION TITLE
FortrezZ WV-01 water valve lost Lifeline association

### DISCUSSION THREAD

**Discussion by denix0** [zwave-js/zwave-js#7004] [Request Support / Investigate Issue]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Manual operation of FortrezZ WV-01 water valve recently stopped showing correct status in the UI (used to work before). Upon investigation, there's no Lifeline association to the controller. And no way to add it manually, as when selecting Root Endpoint, the Group reports "No data available"

What did you expect to happen instead?
Lifeline association to work and update the status in the UI when operating the valve manually

  2. Click on '...'
  3. Scroll down to '...'

Anything else we should know?

### Software versions

zwave-js-ui: 9.14.5
zwave-js: 12.12.3

### Device information

Manufacturer: FortrezZ LLC
Model name: WV-01
0x0084 0x0213-0x0214

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

Will gladly provide logs, if necessary

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment]
Please re-interview the device. If it still doesn't work afterwards, share the log.

**Reply by denix0** [reply]
I have several of these valves in different setups/networks and all of them started exhibiting this issue... Re-interview was the first thing I tried w/o success.

Anyway, here's the log - the device is Node 015
[zwavejs_2024-07-12.log](https://github.com/user-attachments/files/16198415/zwavejs_2024-07-12.log)

**Reply by AlCalzone** [reply]
According to the log, the devices don't support Association CC at all. Not sure how you could have a lifeline before:
```
2024-07-12T19:07:05.361Z CNTRLR « [Node 015] node info received
                                  supported CCs:
                                  · Binary Switch
                                  · Manufacturer Specific
                                  · Version
                                  · Notification
                                  · Application Status
                                  · Configuration
```

Can you share a log of what happens when you control the device manually?

**Reply by denix0** [reply]
Here you go - opened/closed it few times
[zwavejs_2024-07-12(1).log](https://github.com/user-attachments/files/16199585/zwavejs_2024-07-12.1.log)

**Reply by AlCalzone** [reply]
There's nothing at all in the log. I have no clue how it was ever working. Maybe you had an automation that polled the state regularly?

**Reply by denix0** [reply]
Nothing special - no polling. I had automation that would send notification and turn off corresponding water softener plug when the valve is closed and it used to work immediately when operating manually. Now it doesn't...
According to the manual, it is supposed to support device association:
https://www.homecontrols.com/homecontrols/products/pdfs/FZ-FortrezZ/FZWV01-User-Manual.pdf
Could it be that it does not advertise the capability when interviewed due to this being an old device?

**Reply by AlCalzone** [reply]
That manual is speaking of creating associations on other devices where the valve is the target. That does not mean that the valve supports associations though. You could try sending an Association command to the valve and see if it responds. This requires a [driver function](https://zwave-js.github.io/zwave-js-ui/#/usage/driver_function) and watching the logs. Something like this might work:

```ts
const node = driver.controller.nodes.get(15);
const api = node.createAPI(0x85, false);

await api.getGroupCount();
```

**Reply by denix0** [reply]
Sorry for the delay. All I'm getting is:

```
Success zwave api call driverFunction undefined
```

**Reply by AlCalzone** [reply]
You may be looking at the wrong logs. Enable [driver logs](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs), set them to level "debug". If you don't have "log to file" active there, they should appear in the log window.

**Comment by denix0** [comment]
Sorry for another delay.
Hopefully I'm getting the right logs this time:
```
2024-08-21T19:11:14.316Z DRIVER   all queues busy
2024-08-21T19:11:14.316Z SERIAL » 0x011000a90001000f028505250000000012fd                              (18 bytes)
2024-08-21T19:11:14.316Z DRIVER » [Node 015] [REQ] [SendDataBridge]
                                  │ source node id:   1
                                  │ transmit options: 0x25
                                  │ callback id:      18
                                  └─[AssociationCCSupportedGroupingsGet]
2024-08-21T19:11:14.319Z SERIAL « [ACK]                                                                   (0x06)
2024-08-21T19:11:14.322Z SERIAL « 0x010401a90152                                                       (6 bytes)
2024-08-21T19:11:14.322Z SERIAL » [ACK]                                                                   (0x06)
2024-08-21T19:11:14.322Z DRIVER « [RES] [SendDataBridge]
                                    was sent: true
2024-08-21T19:11:14.337Z SERIAL « 0x011d00a91200000100b77f7f7f7f01010300000000020100007f7f7f7f7f90    (31 bytes)
2024-08-21T19:11:14.337Z SERIAL » [ACK]                                                                   (0x06)
2024-08-21T19:11:14.337Z DRIVER « [REQ] [SendDataBridge]
                                    callback id:            18
                                    transmit status:        OK, took 10 ms
                                    routing attempts:       1
                                    protocol & route speed: Z-Wave, 40 kbit/s
                                    routing scheme:         LWR
                                    ACK RSSI:               -73 dBm
                                    ACK channel no.:        1
                                    TX channel no.:         1
2024-08-21T19:11:15.361Z CNTRLR   [Node 015] Timed out while waiting for a response from the node (ZW0201)
2024-08-21T19:11:15.362Z DRIVER   all queues idle
```

**Reply by AlCalzone** [reply]
Looks like the device does not report any association groups (no support for the CC, no response to queries). So I really have no clue how this ever worked.

**Reply by denix0** [reply]
@AlCalzone
Somehow it is now working again with the latest versions of zwave-js UI and the driver - I noticed when I operate the valve locally, I get a "value updated" event even though the groups associations list is empty.
Were there any recent changes in this area?

**Reply by AlCalzone** [reply]
No 🤷🏻‍♂️
